pkg/kubernetes/helm.go

```diff
@@ -336,77 +314,66 @@ func (h *HelmExecute) RunHelmTest() error {
 		helmParams = append(helmParams, "--debug")
 	}
 
-	h.utils.Stdout(h.stdout)
-	log.Entry().Info("Calling helm test ...")
-	log.Entry().Debugf("Helm parameters: %v", helmParams)
-	if err := h.utils.RunExecutable("helm", helmParams...); err != nil {
+	if err := h.runHelmCommand(helmParams); err != nil {
 		log.Entry().WithError(err).Fatal("Helm test call failed")
 	}
 
 	return nil
 }
 
-// RunHelmRegistryLogin is used to login private registry
-func (h *HelmExecute) RunHelmRegistryLogin() error {
-	helmParams := []string{
-		"registry login",
+//RunHelmPublish is used to upload a chart to a registry
+func (h *HelmExecute) RunHelmPublish() error {
+	err := h.runHelmInit()
+	if err != nil {
+		return fmt.Errorf("failed to execute deployments: %v", err)
 	}
-	helmParams = append(helmParams, "-u", h.config.HelmRegistryUser)
-	helmParams = append(helmParams, h.config.HelmChartServer)
 
-	h.utils.Stdout(h.stdout)
-	log.Entry().Info("Calling helm login ...")
-	log.Entry().Debugf("Helm parameters: %v", helmParams)
-	if err := h.utils.RunExecutable("helm", helmParams...); err != nil {
-		log.Entry().WithError(err).Fatal("Helm push login failed")
+	if len(h.config.TargetChartRepositoryURL) == 0 {
+		return fmt.Errorf("there's no target repository for helm chart publishing configured")
 	}
 
-	return nil
-}
-
-// RunHelmRegistryLogout is logout to login private registry
-func (h *HelmExecute) RunHelmRegistryLogout() error {
-	helmParams := []string{
-		"registry logout",
+	repoClientOptions := piperhttp.ClientOptions{
+		Username:     h.config.TargetChartRepositoryUser,
+		Password:     h.config.TargetChartRepositoryPassword,
+		TrustedCerts: h.config.CustomTLSCertificateLinks,
 	}
-	helmParams = append(helmParams, h.config.HelmChartServer)
 
-	h.utils.Stdout(h.stdout)
-	log.Entry().Info("Calling helm logout ...")
-	log.Entry().Debugf("Helm parameters: %v", helmParams)
-	if err := h.utils.RunExecutable("helm", helmParams...); err != nil {
-		log.Entry().WithError(err).Fatal("Helm push logout failed")
+	h.utils.SetOptions(repoClientOptions)
+
+	binary := fmt.Sprintf("%v", h.config.DeploymentName+"-"+h.config.PackageVersion+".tgz")
+
+	targetPath := fmt.Sprintf("helm/%s/%s", h.config.PackageVersion, binary)
```

Comment:
The target path likely needs to be changed. I am waiting for feedback from an expert.

For now we may just push the package directly into the root, like `targetPath := binary`